pkg/util/pki/generate.go

```diff
@@ -108,16 +108,23 @@ func GenerateECPrivateKey(keySize int) (*ecdsa.PrivateKey, error) {
 }
 
 // EncodePrivateKey will encode a given crypto.PrivateKey by first inspecting
-// the type of key provided.
+// the type of key encoding and then inspecting the type of key provided.
 // It only supports encoding RSA or ECDSA keys.
-func EncodePrivateKey(pk crypto.PrivateKey) ([]byte, error) {
-	switch k := pk.(type) {
-	case *rsa.PrivateKey:
-		return EncodePKCS1PrivateKey(k), nil
-	case *ecdsa.PrivateKey:
-		return EncodeECPrivateKey(k)
+func EncodePrivateKey(pk crypto.PrivateKey, crt *v1alpha1.Certificate) ([]byte, error) {
```

Comment:
Would you mind extending the unit tests here to cover this function?

Hardcoding a PEM encoded certificate, decoding it into a crypto.PrivateKey and then encoding with the 3 different values KeyEncoding can be, and verifying their output, would be great 😄

Reply:
Hi @munnerz, I've created the unit test in [generate_test.go](https://github.com/jetstack/cert-manager/blob/76e91e3f2cb6a17715e27b137762a063faff8d36/pkg/util/pki/generate_test.go#L340-L452). Is this the correct place to put the unit test? Also, have I created the unit test properly?